packages/plugins/GeoLocation/src/store/actions.ts

```diff
@@ -130,11 +131,10 @@ const actions: PolarActionTree<GeoLocationState, GeoLocationGetters> = {
       configuredEpsg
     )
 
-    const boundaryCheckPassed = await passesBoundaryCheck(
-      map,
-      boundaryLayerId,
-      transformedCoords
-    )
+    const boundaryCheckPassed =
+      typeof boundaryLayerId === 'string'
+        ? await passesBoundaryCheck(map, boundaryLayerId, transformedCoords)
+        : containsCoordinate(map.getView().calculateExtent(), transformedCoords)
```

Comment:
According to the [OL docs](https://openlayers.org/en/latest/apidoc/module-ol_View-View.html#calculateExtent), this only addresses the currently visible map's view's extent, and not the map's whole extent, which does not meet the intended functionality change.

Reply:
Adjusted 489d2f020a531a3df206f0496db7799668b84937